src/utils/duration.js

```diff
@@ -39,6 +40,16 @@ function toDoubleDigits(num) {
     }
 }
 
+export function milisecondsToDHMS(numericDuration) {
+    const ms = numericDuration || 0;
+    const d = Math.floor(normalizeAge(ms / ONE_DAY));
+    const h = Math.floor(normalizeAge((ms % ONE_DAY) / ONE_HOUR));
+    const m = Math.floor(normalizeAge((ms % ONE_HOUR) / ONE_MINUTE));
+    const s = Math.floor(normalizeAge((ms % ONE_MINUTE) / ONE_SECOND));
+
+    return `+ ${d > 0 ? d + "d " : ""}${h > 0 ? h + "h " : ""}${ m > 0 ? m + "m " : ""}${s > 0 ? s + "s " : ""}`;
```

Comment:
Can you create another utility function `addTimeSuffix(value, suffix) {...}` where you pass in the number and the suffix, it'll check if it's greater than zero, then return the value plus suffix if so or just the value if not. So for days it would be: `const d = addTimeSuffix(Math.floor(normalizeAge(ms / ONE_DAY)), 'd ');` then you don't have to user the tertiary operators in the returned string.